src/components/v2/MyAccount/MyAccount.tsx

```diff
@@ -0,0 +1,106 @@
+/** @jsxImportSource @emotion/react */
+import React, { useState } from 'react';
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import { ISliderProps, Slider } from '../Slider';
+import { useMyAccountStyles as useStyles } from './MyAccountStyles';
+import { Toggle } from '../Toggle';
+import { Icon } from '../Icon';
+import { Tooltip } from '../Tooltip';
+
+const myAccountResources = {
+  myAccountTitle: 'My account',
+  apyWithXvs: 'APY with XVS',
+  apyWithXvsTooltip: 'tooltip content',
+  netApy: 'Net APY',
+  dailyEarnings: 'Daily earnings',
+  supplyBalance: 'Supply balance',
+  borrowBalance: 'Borrow balance',
+  borrowLimit: 'Borrow limit:',
+  safeLimit: 'Your safe limit:',
+};
+
+interface IMyAccountProps {
+  netApy: string;
+  dailyEarnings: string;
+  supplyBalance: string;
+  borrowBalance: string;
+  borrowLimit: string;
+  borrowLimitSum: string;
+  safeLimit: string;
+  sliderProps: ISliderProps;
+}
+
+export const MyAccount = ({
+  netApy,
+  dailyEarnings,
+  supplyBalance,
+  borrowBalance,
+  borrowLimit,
+  borrowLimitSum,
+  safeLimit,
+  sliderProps,
+}: IMyAccountProps) => {
+  const [isToggleSwitched, setToggleSwitched] = useState(true);
+  const handleSwitch = (event: React.ChangeEvent<HTMLInputElement>, checked: boolean) => {
```

Comment:
I'd suggest importing the types of the `Toggle` component and assigning the type of its `onChange` property to `handleSwitch`.

Reply:
could you explain in more details?  
you suggest to pass callback for state management like setToggleSwitched into Toggle onChange? or I missed how should we manage component state

Reply:
For sure. Your implementation is correct, I just mentioned you could reuse the type of `IToggleProps`:
```
const handleSwitch: IToggleProps['onChange'] = (event, checked) => {
  ...
}
```

This way if the type of the `onChange` property of the `Toggle` component changes, `MyAccount` will use the updated type automatically.

Reply:
oh. misunderstood. sure. gonna add typing